app/models/assignment.rb

```diff
@@ -41,7 +41,7 @@ def single_visual_result_html
   end
 
   def results_visible?
-    visible_success_output? || should_retry?
+    (visible_success_output? || should_retry?) && !exercise.choices?
```

Comment:
¿May this fail on interactive problems?

Reply:
Interactive problems inherit the choices field from Exercise, so this should not be a problem.